.github/workflows/PR.yml

```diff
@@ -241,8 +207,90 @@ jobs:
               ) || true
 
       - name: Archive packages
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: osx-packages
           path: |
             /tmp/artifacts/packages
+
+  build_and_test-osx-arm64:
+    name: OSX-ARM64 Tests
+    runs-on: macOS-14 # M1
+    strategy:
+      fail-fast: true
+      max-parallel: 4
+    needs: build-linux
```

Comment:
What is the idea to wait for the Linux job ? It behaves the same at Azure, I just don't know why

Reply:
Historically, this was because there were more limitations on parallel OSX jobs. Not sure if that is still the case and which CI platforms it applies to. Another reason it was kept around was to shake out any issues on Linux first rather than waste time on two failed builds.

Reply:
Going to leave as-is for now. This CI is manually disabled anyway and I'm just updating it for ease of future use. We could remove from here and Azure in another PR after reviewing the current state of limitations.